[master < T2587-PL] Add changelog release notes for Lab v2.5.0 (#750)
memgraph/docs: lab/changelog.md

sidebar_label: Changelog
---

## Lab v2.5.0 - <TODO: Add the release date>

## What's New

> * In case of multiple Cypher queries in the query editor, you will be able to select a single query and run
>   just that one without a need to comment out all the other Cypher queries.

Reviewer: ```suggestion
* If there are several Cypher queries in the query editor you can select a single query and run
  it without commenting out all the other queries.
```

<TODO: Add screenshot here>

> * A limit of having one query execution view has been removed and you can have multiple query executions now. It makes
>   life easier especially when you want to compare different Cypher queries side to side in terms of speed or
>   query results.

Reviewer: ```suggestion
* You can now open multiple query executions views and compare execution speed or results of Cypher queries side by side.
```

Author: Hahah this is great, my complex 3 lines were replaced with a simple 1-2 liner. :D

<TODO: Add screenshot here>

> * Showing only the last result when running multiple Cypher queries at once has been changed. Now, results for each
>   Cypher query will be returned and you will be able to select and view each one.

Reviewer: ```suggestion
* If you are running multiple Cypher queries at once, you can now view the result of each one instead of viewing just the last result. 
```

<TODO: Add screenshot here>

> * When using code blocks in rich collection descriptions, you can specify one of the following languages to have
>   syntax highlighting enabled: `cypher`, `bash`, `python`, `css`, `c`, `cpp`, `json`, `sql`, and `yaml`. Check the
>   examples for syntax highlighting on [Markdown Guide](https://www.markdownguide.org/extended-syntax/#syntax-highlighting).

Reviewer: ```suggestion
* You can highlight the syntax of code blocks in the rich collection description by using one of the following language styles: `cypher`, `bash`, `python`, `css`, `c`, `cpp`, `json`, `sql`, and `yaml`. Check the
  examples of syntax highlighting in the [Markdown Guide](https://www.markdownguide.org/extended-syntax/#syntax-highlighting).
```

Reviewer: BTW the interface does not refere this as RICH collections but query collections

<TODO: Add screenshot here>

> * Query modules are now sorted alphabetically which will enable easier and faster. A search has been added for query
>   module procedures if there are more than 5 procedures within a module for easier narrowing down (e.g. `nxalg`
>   with [49 procedures](https://memgraph.com/docs/mage/query-modules/python/nxalg)).

Reviewer: ```suggestion
* Query modules are now sorted alphabetically for easier and faster browsing. A search box has also been added to query modules with more than 5 procedures to help narrow them down (e.g. `nxalg` query module has [49 procedures](https://memgraph.com/docs/mage/query-modules/python/nxalg)).
```

> * Rendering graph view for medium/larger graphs might take time, so for any graph with more than 5k nodes or 10k edges
>   you will be asked if you want to proceed with the graph visualization or switch to the data view. This also fixed
>   showing a default graph view for each new query execution and accidentally rendering the full database graph.

Reviewer: ```suggestion
* When rendering a graph with more than 5k nodes or 10k relationships, which might take considerable amount of time to preview, you will be asked if you want to proceed with the graph visualization or switch to the data view. Due to this feature, graph view is no longer a default result view. 
```

Reviewer: I'm not actually sure what this means: This also fixed
  showing a default graph view for each new query execution and accidentally rendering the full database graph.

Author: I will take your suggestion, but I will remove the last sentence "Due to this feature, graph view is no longer a default result view" because it is not entirely true.

To sync with you on that so you are aware of how it works:

* When you open query execution and run a query such as `MATCH (n)-[e]->(m) RETURN n, e, m` - this query returns nodes and relationships which means "Graph view" will be shown and it will start rendering the graph (the graph might be 1M nodes and 1M relationships)
* When you open query execution and run a query such as `MATCH (n)-[e]-(m) RETURN n.name` - this query returns a primitive, name. The default view will be "Data view" because Lab can't show a graph view as there are no nodes and relationships in the result.
* Within this query execution, if the user switches between Data and Graph view, that view will be the one that is favored. E.g. user switches to Data view and runs another query that returns nodes and relationships (Graph view can render it), the Lab will still show the last user selection, which was Data view.

The feature above will just add a protection layer for rendering large graphs and waiting for it to complete the render. So whenever graph view is selected and there is a bunch of nodes and relationships there will be an action: "Are you sure you want to proceed". 
Just having this will also fix issues that users had where they would have millions of nodes/edges and they would open new query execution and run `MATCH (n)-[e]->(m) RETURN n, e, m` and then rendering starts which takes a lot of time for millions of nodes/relationships.

Reviewer: I also agree you can just leave that last sentence out! :)

* Changes in the Cypher query editor and GSS style editor will be automatically saved in rich collections. You will

>   also have the option to do the manual save like before.

Reviewer: * In rich collections, you can now choose between saving changes in the Cypher query editor and GSS style editor manually or automatically.

* Besides manually saving changes in the Cypher query and GSS style editor, they will also be saved automatically.

I'm again not sure what the feature is... if they are saved automatically, why would I save them manually as well? Or do I choose between manual save and autosave?

Reviewer: yes, perfect! :)

> * Notifications have been added to the Lab where you will be able to receive information about new product updates,
>   and various tips and tricks around the Memgraph ecosystem.

Reviewer: ```suggestion
* Memgraph Lab will now notify you of any product updates and offer various tips and tricks for using the Memgraph ecosystem.
```

Author: I really have issues with the passive voice vs active (better one) that you add. I will need to go through that Google document for Tech writing again.

## Bug Fixes

* Cypher intellisense for node/edge labels and properties was working for graphs with a maximum of 100k nodes and
  200k edges. The limits have been increased so labels and properties intellisense will work for graphs with a maximum

>   of 250k nodes and 500k edges.

Reviewer: Cypher code suggestions can now handle labels and properties of 250k nodes and 500k relationship, compared to the previous limit of 100k nodes nad 200k relationships.

> * Scrolling in the rich collection was hard because of multiple scrollable elements such as query editor and query
>   results. Now, on click, you can focus a particular segment and the scroll will only work within that focused segment.

Reviewer: ```suggestion
* 
Multiple scrollable elements of the rich collection was making scrolling difficult. Now you can focus on a particular element and scroll through it by clicking on it.
```

Author: How neat, simple, yet great explanation :D

* Browser's back button is now working as expected when using Lab as a web application.

> * Not showing graph schema on a database with no edges has been fixed.

Reviewer: ```suggestion
* Graph schema is now generated even if the database has no relationships.
```

> * Missing in-progress feedback when generating graph schema and exporting datasets for graphs with more than 10M nodes
>   have been fixed.

Reviewer: ```suggestion
* In-progress feedback when generating graph schema and exporting datasets for graphs with more than 10M nodes
  is now previewed as expected.
```

> * A glitch with a lot of empty spaces when using the search on Lab datasets has been fixed.

Reviewer: ```suggestion
* Dataset cards no longer spread apart when conducting a search. 
```
